### ISSUE TITLE
Tap callbacks in PhotoViewGalleryPageOptions

### ISSUE THREAD

**Issue by hugocbpassos** [renancaraujo/photo_view#146]
Like `PhotoView`, `PhotoViewGalleryPageOptions` should have tap callbacks (`onTapUp` and `onTapDown`).

Giving a context, the gallery of my app shows/hides `AppBar` on user tap, just like WhatsApp.

**Comment by renancaraujo** [comment]
Great catch!
That should not be complex tackle.

**Comment by renancaraujo** [comment]
A fix for that has been included on master, since photoview 0.4.0 is not yet beta compatible, I wont publish any release for now. However, you can depend directly on our master to get that enhancement.

**Comment by hugocbpassos** [comment]
@renancaraujo Flutter has reached 1.7.8 in stable channel. Perhaps you should consider publishing a new release.

**Comment by hugocbpassos** [comment]
@renancaraujo Also, I tried depend directly on master and https://github.com/renancaraujo/photo_view/issues/150 seems to be fixed, but the callbacks that you added to fix this issue are never called.

You can run the code below and check that "onTapUp" and "onTapDown" are never printed:

```dart
import 'package:flutter/material.dart';
import 'package:photo_view/photo_view_gallery.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(body: Foo()),
    );
  }
}

class Foo extends StatefulWidget {
  @override
  _FooState createState() => _FooState();
}

class _FooState extends State<Foo> {
  static const images = <String>[
    'https://homepages.cae.wisc.edu/~ece533/images/peppers.png',
    'https://homepages.cae.wisc.edu/~ece533/images/baboon.png',
    'https://homepages.cae.wisc.edu/~ece533/images/girl.png',
  ];

  @override
  Widget build(BuildContext context) {
    return PhotoViewGallery.builder(
      itemCount: images.length,
      builder: (_, int index) {
        return PhotoViewGalleryPageOptions(
          imageProvider: NetworkImage(images[index]),
          onTapUp: (_, __, ___) => print('onTapUp'),
          onTapDown: (_, __, ___) => print('onTapDown'),
        );
      },
    );
  }
}
```

**Comment by pitayane** [comment]
> @renancaraujo Also, I tried depend directly on master and #150 seems to be fixed, but the callbacks that you added to fix this issue are never called.

Because it's not passed to the PhotoView.
See photo_view_gallery.dart  around line 254.

![image](https://user-images.githubusercontent.com/4426237/61112245-fc4a1b00-a4bd-11e9-8bab-442710cfb842.png)

I have the exactly same problem here.
Hope it can help u.

**Comment by renancaraujo** [comment]
Supposed to be solved on 0.4.2
